smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ExamplesTraitValidator.java

```diff
@@ -53,23 +53,34 @@ private List<ValidationEvent> validateExamples(Model model, OperationShape shape
                 events.addAll(input.accept(validator));
             });
 
-            model.getShape(shape.getOutputShape()).ifPresent(output -> {
-                NodeValidationVisitor validator = createVisitor(
-                        "output", example.getOutput(), model, shape, example);
-                events.addAll(output.accept(validator));
-            });
+            boolean isOutputDefined = example.getOutput().isPresent();
+            boolean isErrorDefined = example.getError().isPresent();
+
+            if (isOutputDefined && isErrorDefined) {
+                events.add(error(shape, trait, String.format(
+                        "Example: `%s` has both output and error defined, only one should be present.",
+                        example.getTitle())));
+            } else {
+                if (isErrorDefined) {
```

Comment:
nit: the `if` statements for `isErrorDefined` and `isOutputDefined` can be moved into the outer layer as `else if` statements.